### DISCUSSION TITLE
Syntax highlighting.

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [damionx7/Zettel-Notes-Documentation#135] [Feature Requests]
This doesn't necessarily necessitate that it be able to differentiate between languages, since currently it solely displays `.MD` files, per https://github.com/damionx7/Zettel-Notes-Documentation/discussions/70#discussion-6836176.

**Comment by damionx7** [comment]
Can you explain this feature request a bit more or maybe elaborate the title.

**Reply by roke-julian-lockhart** [reply]
@damionx7, I'd like [this](https://github.com/mxstbr/markdown-test-file/blob/master/TEST.md?plain=1#L108C1-L115C59):

```MD
> ## This is a header.
> 
> 1.   This is the first list item.
> 2.   This is the second list item.
> 
> Here's some example code:
> 
>     return shell_exec("echo $input | $markdown_script");
```

See how it's highlighted? It makes a massive difference when editing markup, just like it does when programming. Surely your code editor uses SH?

**Reply by damionx7** [reply]
Have you tried changing Markdown theme in Settings > Theme > Markdown Style.

I have added an example of a theme configured to use green color as text color and blue color as heading color.

![Screenshot_2024-07-24-22-48-22-19_377fe8767025e4fa6dddb62d6df8f716](https://github.com/user-attachments/assets/9752b189-345f-4182-b1ef-5446ab4b1626)

**Reply by roke-julian-lockhart** [reply]
> https://github.com/damionx7/Zettel-Notes-Documentation/discussions/135#discussioncomment-10140730

@damionx7, I have it set to follow the system, so I can't modify that preference.

**Reply by damionx7** [reply]
Sure. You can modify it. Just uncheck the follow system option and then set markdown styles for day and night both. And then you can check the follow system option.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/damionx7/Zettel-Notes-Documentation/discussions/135#discussioncomment-10140859

Ah, I expected that it would revert when "Follow system" is enabled again. In which case, why is the option disabled when "Follow system" is enabled, if it doesn't affect it...?

**Reply by damionx7** [reply]
You can set different theme settings for day and night mode. Thats why these options are disabled when follow system is checked. For eg. you can have different markdown style, font for day and night mode.

**Reply by damionx7** [reply]
Closing this discussion for now.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/damionx7/Zettel-Notes-Documentation/discussions/135#discussioncomment-10143448

@damionx7, how does one set those, then? Does the user have to change their OS theme to do so?

**Reply by damionx7** [reply]
Please see this video for instructions on how to set different markdown style for light and dark mode.

https://github.com/user-attachments/assets/dfb8c0fc-7056-438e-9cb5-dfeae61eb115

**Reply by roke-julian-lockhart** [reply]
> https://github.com/damionx7/Zettel-Notes-Documentation/discussions/135#discussioncomment-10157799

Thanks, @damionx7. So do those custom modifications apply to both night and dark mode, or merely one? I expected that they would apply to merely the light or dark mode selected at the moment that they were customised, but it appears in that video that they affect both.

This is quite a powerful feature which doesn't necessarily have the most intuitive GUI. Some documentation of it, I think, would be of significant use.

**Reply by damionx7** [reply]
When you select light mode and make the modifications, then these modifications are saved for light mode. And whenever you select light mode or if light mode is activated from the follow system option, then these modifications are applied. Vice versa for dark mode. 

Feel free to contribute to the documentation (this repo). I usually think from a developer's perspective while writing documentation and miss some areas such as these theme option and the perspective of a user is much needed to cover these areas.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/damionx7/Zettel-Notes-Documentation/discussions/135#discussioncomment-10251683

@damionx7, thanks. That explains it well. I'll try to contribute something.
